source/data-modeling/associations.txt

```diff
@@ -0,0 +1,517 @@
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-belongs-to
+   :end-before: # end-belongs-to
+
+You can allow {+odm+} to persist documents to the database without storing the
+``_id`` of the associated parent class by setting the ``optional`` option to
+``true``, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-belongs-to-optional
+   :end-before: # end-belongs-to-optional
+
+.. tip::
+
+   You can globally change the default behavior of the ``belongs_to``
+   association to not require their parent class by setting the
+   ``belongs_to_required_by_default`` configuration option to ``false`` in your
+   application's configuration settings.
+
+You can specify a ``belongs_to`` association in a child class without specifying a matching
+``has_one`` or ``has_many`` association in the parent class. When doing so, you
+can't access the fields of the child document from the parent class, but you can
+access the parent fields that are stored in the child class, such as the
+parent's ``_id`` field. In the following example, the ``Band``
+class cannot access the ``Members`` class, but the ``Members`` class can access the
+``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-belongs-to-one-way
+   :end-before: # end-belongs-to-one-way
+
+For clarity, you can optionally set the ``inverse_of`` option to ``nil`` to
+indicate that the parent class does not contain a ``has_one`` or ``has_many``
+association to the child class, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-belongs-to-inverse
+   :end-before: # end-belongs-to-inverse
+
+Has and Belongs To Many
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the ``has_and_belongs_to_many`` macro to declare that a class model contains
+a many-to-many relationship with another class. In a many-to-many relationship,
+each document in one class can be associated with multiple documents in another
+class. The following example creates a ``Band`` class with a
+``has_and_belongs_to_many`` relationship to a ``Members`` class. A ``Band`` document can
+reference multiple ``Members`` documents, and a ``Members`` document can
+reference multiple ``Band`` documents.
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-has-and-belongs-to-many
+   :end-before: # end-has-and-belongs-to-many
+
+When you declare a ``has_and_belongs_to_many`` association, both model instances
+store a list of the associated document's ID values. You can set the
+``inverse_of`` option to ``nil`` to store the associated document's ID values in
+only one of the model instances. The following example prompts {+odm+} to store
+the associated document's ID values in only the ``Band`` class:
```

Comment:
S: since you use `_id` in other mentions
```suggestion
store a list of the associated document's ``_id`` values. You can set the
``inverse_of`` option to ``nil`` to store the associated document's ``_id`` values in
only one of the model instances. The following example prompts {+odm+} to store
the associated document's ``_id`` values in only the ``Band`` class:
```

Reply:
Applies to all